src/Functions/FunctionBinaryArithmetic.h

```diff
@@ -1209,6 +1213,17 @@ class FunctionBinaryArithmetic : public IFunction
             return arguments[0];
         }
 
+        /// Special case - one or both arguments are IPv4
+        if (isIPv4(arguments[0]) || isIPv4(arguments[1]))
```

Comment:
Needs a test, I am almost sure that it's always false.

Reply:
Actually, function isIPv4 is not connected with the template above :) It's
`inline bool isIPv4(const DataTypePtr & data_type) { return WhichDataType(data_type).isIPv4(); }` so should work aifak.
Test is added, I will merge.